crates/polars-core/src/series/mod.rs

```diff
@@ -372,6 +372,11 @@ impl Series {
     pub fn cast_with_options(&self, dtype: &DataType, options: CastOptions) -> PolarsResult<Self> {
         use DataType as D;
 
+        #[cfg(feature = "dtype-array")]
+        if matches!(dtype, D::Array(_, width) if *width == 0) {
```

Comment:
Can we fail a bit lower? Maybe in the array cast?

Reply:
I started looking yesterday at just implementing zero-width arrays. It is almost done so this PR is not really needed anymore.

Reply:
Alright. Even better. :P